app/domain/authentication/constraints/non_permitted_constraint.rb

```diff
@@ -0,0 +1,22 @@
+module Authentication
+  module Constraints
+
+    # This constraint is initialized with an array of strings.
+    # They represent resource restrictions that are not allowed
+    # Calling `validate` enforces this constraint on the given list of resource restrictions.
+    # If there is annotation for one of these non permitted values proper error would be thrown
+    class NonPermittedConstraint
```

Comment:
@tzheleznyak we will have more functions in this class? if not, why not making it command class?

Reply:
To be in the same form as other constraints. If im turning this to CommandClass it can be in separate issue to refactor all validators . 

And it will influence also GCP, OIDC, K8S